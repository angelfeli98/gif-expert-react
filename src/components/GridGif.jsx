
import PropTypes from 'prop-types';

import Gif from './Gif';
import useFetchGifs from '../hooks/useFetchGifs';

function GridGif({category}) {

  const  [isLoading, gifs] = useFetchGifs(category);

  return (
    <>
      { isLoading && <h2> ...loading </h2> }

      { !isLoading && <div>
          <h2> {category} </h2>
            <ul className="card-grid" >
                {
                    gifs.map(gif => <Gif key={gif.id} gif={gif}/>)
                }
            </ul>
        </div>
      }
    </>
  )
}

GridGif.propTypes = {
    category: PropTypes.string.isRequired
}

export default GridGif;