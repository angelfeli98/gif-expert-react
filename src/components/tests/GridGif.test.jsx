/* eslint-disable no-undef */

import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';

import GridGif from '../GridGif';

jest.mock('../../utils/env', () => ({key: 'qPsZQJFTytqz0J6TqTmCZb58dTWaqyRX', baseUrl: 'https://api.giphy.com/v1/gifs/search'}));


describe('<GridGif />', () => {
    test('should render correctly', () => {
        const component = render(<GridGif category="halo" />);
        expect(component.container).toMatchSnapshot();
    })

    test('should render loading text', () => {
        const component = render(<GridGif category="halo" />);
        expect(component.getByText('...loading')).toBeDefined();
    })

    test('should render gifs', async() => {
        render(<GridGif category="halo" />);
        await waitFor(() => {
            expect(screen.getAllByRole('listitem').length).toBe(10);
        });
    })

    test('should not render gifs', async() => {
        render(<GridGif category="kjwebhorgtiwur4htogiuwererwtherthy" />);
        await waitFor(() => {
            expect(screen.queryAllByRole('listitem')).toEqual([]);
        });
    })
})