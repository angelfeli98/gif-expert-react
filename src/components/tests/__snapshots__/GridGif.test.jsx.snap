// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridGif /> should render correctly 1`] = `
<div>
  <h2>
     ...loading 
  </h2>
</div>
`;
